library/src/main/java/com/novoda/downloadmanager/CallbackThrottleCreator.java

```diff
@@ -41,7 +41,7 @@ private CallbackThrottleCreator(Type type, TimeUnit timeUnit, long frequency, Cl
     CallbackThrottle create() {
         switch (type) {
             case THROTTLE_BY_TIME:
-                return new CallbackThrottleByTime(timeUnit.toMillis(frequency));
+                return new CallbackThrottleByTime(SchedulerFactory.createFixedRateTimerScheduler(timeUnit.toMillis(frequency)));
```

Comment:
i'd make extract out the Scheduler creation into a local var. To make it clear about what is being created.